### PR TITLE
Add urlFor to templates

### LINKED ISSUE #21: Fix URLs
Add a `get_url` equivalent function and fix navbar links, just set models `urls` to `WebApplication.URLS` or `WebApplication.URLS.class`.

### PR DESCRIPTION
**Pull request by Herover**
Closes #21

### CODE REVIEW

**Comment by smilladion** [comment]
Not sure how big the scope of this PR is, but some links are still weird. After you are logged in, if you click "my timeline" then it goes to the public timeline and the navbar link for "my timeline" disappears (but it comes back again when you click "public timeline"... quite strange).

Trying to post a message gives a NullPointerException. Although maybe that error is part of a different issue.

Signing up as a new user gives SQLiteException: Database is locked. Not sure what it means... maybe something on my end?

Now I'm trying to sign in again and getting 500 Internal Server Error... very weird. All I did was replace the database with one that had a few messages in it. Maybe that makes it break more.

Also this for the username:
image

File: minitwit/src/main/java/WebApplication.java [hunk 1]
```diff
@@ -39,6 +39,34 @@
...
+                } else {
+                    System.out.println("Cant convert value, lets see what happens now... " + key + " : " + values.get(key));
+                    u = u.replace(":" + key, "" + values.get(key));
+                }
+            }
+
+            return u;
+        }
+
+        public static String urlFor(String url) throws Exception {
```
Comment (smilladion) [inline minitwit/src/main/java/WebApplication.java:67]: What's the point of this method? It doesn't seem to do anything. :P

Comment (Herover) [reply minitwit/src/main/java/WebApplication.java:67]: It allows calling urlFor without url parameters like [this](https://github.com/Herover/itu-devops-h/pull/47/files#diff-1b1a0c14b12b0486b3a2b443c33368874d5d5bfe21a01a8752f84153310beae2L32).

Comment (smilladion) [reply minitwit/src/main/java/WebApplication.java:67]: But at that point, like when you say `urls.urlFor($ADD_MESSAGE)`, is it not the same to simply write `$ADD_MESSAGE`?

**Comment by Herover** [comment]
Fixed the navbar issue.
The post message error is on a problem in add_message using sessions wrong.
I couldn't reproduce the database is locked error on my computer is weird, I can't make it happen on my computer. Is it possible you have a lock file or have the db open in some other program? Hopefully it was just a weird fluke but maybe someone else can test it too.

File: minitwit/src/main/java/WebApplication.java (hunk 1 above)
Comment (Herover) [reply minitwit/src/main/java/WebApplication.java:67]: It is :D

**Comment by smilladion** [comment]
After recreating my database from scratch, I can sign up a new user and log in. But then if I log out and try to create another user, that's when the database error seems to occur. Maybe try that? I don't have it open anywhere else.

**Comment by Herover** [comment]
Wow good find! Is it ok if I open a separate issue for it?

**Comment by smilladion** [comment]
Yeah sure, it's probably unrelated to this one!

**Pull request merged**